subsystems/TypeHandlerLibrary/src/test/java/org/terasology/persistence/typeHandling/coreTypes/GenericMapTypeHandlerTest.java

```diff
@@ -44,23 +132,69 @@ void testDeserialize() {
     }
 
     @Test
+    @DisplayName("Deserializing valid data with a mismatching value type handler fails deserialization (returns empty `Optional`)")
     void testDeserializeWithMismatchedValueHandler() {
         var th = new GenericMapTypeHandler<>(
                 new StringTypeHandler(),
                 new UselessTypeHandler<>()
         );
 
-        assertThat(th.deserialize(testData)).hasValue(Collections.emptyMap());
+        assertThat(th.deserialize(testData)).isEmpty();
     }
 
     @Test
+    @DisplayName("Deserializing valid data with a mismatching key type handler fails deserialization (returns empty `Optional`)")
     void testDeserializeWithMismatchedKeyHandler() {
         var th = new GenericMapTypeHandler<>(
                 new UselessTypeHandler<>(),
                 new LongTypeHandler()
         );
 
-        assertThat(th.deserialize(testData)).hasValue(Collections.emptyMap());
+        assertThat(th.deserialize(testData)).isEmpty();
+    }
+    
+    @Test
+    @DisplayName("Incorrectly formatted data (without an outer array) fails deserialization (returns empty `Optional`)")
+    void testDeserializeWithObjectInsteadOfArray() {
+        var th = new GenericMapTypeHandler<>(
+                new StringTypeHandler(),
+                new LongTypeHandler()
+        );
+
+        assertThat(th.deserialize(testDataMalformatted)).isEmpty();
+    }
+
+    @Test
+    @DisplayName("Incorrectly formatted data (without a map entry with key \"key\") fails deserialization (returns empty `Optional`)")
+    void testDeserializeWithMissingKeyEntry() {
+        var th = new GenericMapTypeHandler<>(
+                new StringTypeHandler(),
+                new LongTypeHandler()
+        );
+
+        assertThat(th.deserialize(testDataMissingKeyEntry)).isEmpty();
+    }
+
+    @Test
+    @DisplayName("Incorrectly formatted data (without a map entry with key \"value\") fails deserialization (returns empty `Optional`)")
+    void testDeserializeWithMissingValueEntry() {
+        var th = new GenericMapTypeHandler<>(
+                new StringTypeHandler(),
+                new LongTypeHandler()
+        );
+
+        assertThat(th.deserialize(testDataMissingValueEntry)).isEmpty();
+    }
+
+    @Test
+    @DisplayName("A map containing both, correctly and incorrectly formatted data, fails deserialization (returns empty `Optional`)")
+    void testDeserializeWithValidAndInvalidEntries() {
+        var th = new GenericMapTypeHandler<>(
+                new StringTypeHandler(),
+                new LongTypeHandler()
+        );
+
+        assertThat(th.deserialize(testDataValidAndInvalidMix)).isEmpty();
```

Comment:
This is the case I'm a little worried about, since it's a change from the previous behavior. Not that the previous behavior was _better,_ but it's one of those things where something might have been implicitly depending on it.

But this has a test case now, and the hypothetical I'm worrying about doesn't, so this wins.

Reply:
I don't think (or hope) something was depending on it. Rather, we'll see some things either work out better or even worse than before, hopefully just epxosing the underlying problem better.

Reply:
My expectation is that if this makes something work worse than before, then it's mainly making the underlying issue more visible than before.

> I'm curious: which case was it that you initially bumped in to and prompted this PR?

I ran into the crash after making `ChangingBlocks` use uris instead of strings for the map key.
The prefabs in PlantPack that I tested this were working before with the basic string key type map deserialization but didn't have the format required by the generic map type handler (outer array, entries with "key" and "value" keys). So the three malformatting test cases are covering this now.